pages/index.tsx

```diff
@@ -1,8 +1,39 @@
-import Head from 'next/head';
+import { useEffect, useState } from 'react';
 import type { NextPage } from 'next';
+import Link from 'next/link';
+import Head from 'next/head';
+import Post from '../components/Post';
 import styles from '../styles/Home.module.css';
 
+interface PostProps {
+  postID: string;
+  author: {
+    userName: string;
+  };
+  title: string;
+  url: URL;
+  description: string;
+  upvotes: number;
+  timeCreated: number;
+}
+
 const Home: NextPage = () => {
+  const [posts, setPosts] = useState<PostProps[]>([]);
+
+  useEffect(() => {
+    const getAllPosts = async () => {
```

Comment:
Scoping

Reply:
@codingknite you're doing the same `getAllPosts` everywhere, it seems like the endpoint `/api/getAllPosts` gets called a lot. Just abstract it into a single function, avoid technical debt